device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -108,13 +110,25 @@ __global__ void find_tracks(
         shared_candidates_size);
 }
 
+/// CUDA kernel for running @c traccc::device::get_sort_key_value
+__global__ void get_sort_key_value(
+    bound_track_parameters_collection_types::const_view params_view,
+    vecmem::data::vector_view<device::sort_key> keys_view,
+    vecmem::data::vector_view<unsigned int> ids_view) {
+
+    int gid = threadIdx.x + blockIdx.x * blockDim.x;
+
+    device::get_sort_key_value(gid, params_view, keys_view, ids_view);
```

Comment:
I would do:

```suggestion
    device::get_sort_key_value(threadIdx.x + blockIdx.x * blockDim.x, params_view, keys_view, ids_view);
```

The new function expects an `std::size_t` value for the "global ID". But here you produce currently a signed integer instead. Worse yet, you produce the signed integer out of `unsigned int` values. (The CUDA types are `unsigned int` I believe...)

Long story short, it seems better to just not use an explicit type here, and just feed the value to the function with this formalism. :thinking:

Reply:
The CUDA values are actually signed integers, so the `int` signature is correct. :cry:

Reply:
Idk, I will just put the id to the function directly as suggested. Probably other functions will also need to be fixed